cmd/lifecycle/restorer.go

```diff
@@ -61,13 +73,20 @@ func (r *restoreCmd) Exec() error {
 	if err := verifyBuildpackApis(group); err != nil {
 		return err
 	}
-	cacheStore, err := initCache(r.cacheImageTag, r.cacheDir)
+	cacheStore, err := initCache(r.cacheImageTag, r.cacheDir, r.keychain)
 	if err != nil {
 		return err
 	}
 	return restore(r.layersDir, group, cacheStore)
 }
 
+func (r *restoreCmd) resolveKeychain() (authn.Keychain, error) {
+	if r.cacheImageTag == "" {
+		return authn.DefaultKeychain, nil // fix the keychain
```

Comment:
Don't we need to fix it?

Reply:
It still feels weird that we call `resolveKeychain()` when we don't actually need a keychain. Maybe it's better for each command to have a helper function like `registryImages()` that returns the images being used and if this is non-empty then we resolve the auth.

So something like 
```
	registryImages := a.registryImages()
	if len(registryImages) > 0 {
		var err error
		a.keychain, err = auth.ResolveKeychain(cmd.EnvRegistryAuth, registryImages)
		if err != nil {
			return cmd.FailErr(err, "resolve keychain")
		}
	}
```
in the `Privileges()` function (very similar to how we initialize the docker client).

Reply:
tbh, I don't mind passing in the empty images or getting an essentially empty keychain back.

Reply:
Sounds good. In this case I think we should add a test for `ResolveKeychain()` that verifies empty images produces a valid keychain.

Reply:
We added said test...